Intersect.Server/Maps/MapInstance.cs

```diff
@@ -235,43 +247,56 @@ public void SpawnItem(int x, int y, Item item, int amount, Guid owner)
                 return;
             }
 
-            
+            // if we can stack this item or the user configured to drop items consolidated, simply spawn a single stack of it.
+            if (itemBase.Stackable || Options.Loot.ConsolidateMapDrops)
+            {
+                var mapItem = new MapItem(item.ItemId, item.Quantity, item.BagId, item.Bag) {
+                    X = x,
+                    Y = y,
+                    DespawnTime = Globals.Timing.TimeMs + Options.Loot.ItemDespawnTime,
+                    Owner = owner,
+                    OwnershipTime = Globals.Timing.TimeMs + Options.Loot.ItemOwnershipTime,
+                    VisibleToAll = Options.Loot.ShowUnownedItems,
+                    Quantity = amount
 
-            var mapItem = new MapItem(item.ItemId, item.Quantity, item.BagId, item.Bag) {
-                X = x,
-                Y = y,
-                DespawnTime = Globals.Timing.TimeMs + Options.Loot.ItemDespawnTime,
-                Owner = owner,
-                OwnershipTime = Globals.Timing.TimeMs + Options.Loot.ItemOwnershipTime,
-                VisibleToAll = Options.Loot.ShowUnownedItems
-            };
+                };
 
-            if (itemBase.ItemType == ItemTypes.Equipment)
-            {
-                mapItem.Quantity = 1;
-                if (mapItem.StatBuffs != null && item.StatBuffs != null)
+                // If this is a piece of equipment, set up the stat buffs for it.
+                if (itemBase.ItemType == ItemTypes.Equipment)
                 {
-                    for (var i = 0; i < mapItem.StatBuffs.Length; ++i)
-                    {
-                        mapItem.StatBuffs[i] = item.StatBuffs.Length > i ? item.StatBuffs[i] : 0;
-                    }
-                }
-                else if (mapItem.StatBuffs == null)
-                {
-                    Log.Warn($"Unexpected null: {nameof(mapItem)}.{nameof(mapItem.StatBuffs)}");
-                }
-                else
-                {
-                    Log.Warn($"Unexpected null: {nameof(item)}.{nameof(item.StatBuffs)}");
+                    mapItem = SetMapItemStatBuffs(mapItem);
```

Comment:
I know there's a cleaner way to do this by void SetMapItemStatBuffs(ref MapItem mapItem).
But which way is preferrable?

@lodicolo

Reply:
Nevermind, I just put it on MapItem. :)